### PR TITLE
IIIF Auth v2 interaction for PDFs -- with rebase

### PR DESCRIPTION
**Pull request by peetucket**
Fixes https://github.com/sul-dlss/sul-embed/issues/1919 - Implement authorization flow for PDFs.

Same as #2032 with all changes from main as of 1/8/2024 rebased and conflicts fixed...pushed to a new branch for troubleshooting

Adds PDF authorization flow for the new PDF viewer, along with some refactoring to move code that is now used by both the media player and the PDF player to a common component.

### CODE REVIEW

File: app/viewers/embed/viewer/pdf_viewer.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def importmap
       end
 
       def stylesheet
-        'pdf.css'
+        'media.css' # TODO: change this
```
Comment (jcoyne) [inline app/viewers/embed/viewer/pdf_viewer.rb:18]: try `companion_window.css`

File: app/assets/stylesheets/media.scss [hunk 2]
```diff
@@ -112,42 +112,6 @@
...
-
-  .dismissButton {
-    padding: 0.2rem;
-    svg {
-      font-size: 1rem;
-      vertical-align: bottom;
-    }
-  }
-}
-
```
Comment (peetucket) [inline app/assets/stylesheets/media.scss:150]: we need this CSS for the login panel, which is also used by the PDF viewer, so move it to the `companion_window` (as shown above)

File: app/components/embed/legacy_pdf_component.html.erb [hunk 3]
```diff
@@ -41,7 +41,7 @@
           <%= javascript_pack_tag('legacy_pdf_viewer') %>
       <% else %>
           <div style="padding: 10%;">
-            <strong><%= I18n.t('restrictions.not_accessibile') %></strong>
+            <strong><%= I18n.t('restrictions.not_accessible') %></strong>
           </div>
```
Comment (peetucket) [inline app/components/embed/legacy_pdf_component.html.erb:45]: typo

File: app/components/embed/login_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div data-file-auth-target="loginPanel" hidden>
+    <div class="authLinkWrapper">
+        <svg class="MuiSvgIcon-root" focusable="false" aria-hidden="true" viewBox="0 0 24 24"><path d="m14.5 14.2 2.9 1.7-.8 1.3L13 15v-5h1.5v4.2zM22 14c0 4.41-3.59 8-8 8-2.02 0-3.86-.76-5.27-2H4c-1.15 0-2-.85-2-2V9c0-1.12.89-1.96 2-2v-.5C4 4.01 6.01 2 8.5 2c2.34 0 4.24 1.79 4.46 4.08.34-.05.69-.08 1.04-.08 4.41 0 8 3.59 8 8zM6 7h5v-.74C10.88 4.99 9.8 4 8.5 4 7.12 4 6 5.12 6 6.5V7zm14 7c0-3.31-2.69-6-6-6s-6 2.69-6 6 2.69 6 6 6 6-2.69 6-6z"></path></svg>
+        <p data-file-auth-target="loginMessage" class="loginMessage">Stanford users: log in to access all available features.</p>
+        <button data-file-auth-target="loginButton" data-action="file-auth#login"
+                 data-file-auth-messageId-param="" data-file-auth-url-param="">
+                Log in
+        </button>
```
Comment (peetucket) [inline app/components/embed/login_component.html.erb:8]: note: the actual text shown in this UI will be updated via javascript to the messages that come back from the auth service (via PURL)

File: config/locales/en.yml [hunk 5]
```diff
@@ -22,8 +22,7 @@
 en:
   hello: "Hello world"
   restrictions:
-    not_accessibile: 'This item cannot be accessed online. See Access conditions for more information.'
-    not_supported_html: 'Your browser does not support viewing PDFs. Please %{download_link} to view it.'
+    not_accessible: 'This item cannot be accessed online. See Access conditions for more information.'
     download_the_file: 'download the file'
```
Comment (peetucket) [inline config/locales/en.yml:26]: 1. fix typo
2. the other translation is now in a JS file above

**Review by jcoyne — APPROVED** [review]

**Pull request merged**